From ef65ae1f14ec9262c7bbf8eac7368d74cf9acd26 Mon Sep 17 00:00:00 2001
From: Zhang Xiao <xiao.zhang@windriver.com>
Date: Wed, 15 Feb 2012 17:04:22 +0800
Subject: [PATCH] preempt_rt/powerpc: Accomplish kunmap_virt with HIGHMEM disabled

kunmap_virt was used for preempt_rt and was only accomplished with HIGHMEN
enabled code. This will cause build error when HIGHMEM disabled. So accomplish
it in HIGHMEM disabled code.

Signed-off-by: Zhang Xiao <xiao.zhang@windriver.com>
---
 arch/powerpc/include/asm/pgtable-ppc32.h |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/include/asm/pgtable-ppc32.h b/arch/powerpc/include/asm/pgtable-ppc32.h
index 2567d67..54d03a7 100644
--- a/arch/powerpc/include/asm/pgtable-ppc32.h
+++ b/arch/powerpc/include/asm/pgtable-ppc32.h
@@ -11,7 +11,14 @@
 extern unsigned long va_to_phys(unsigned long address);
 extern pte_t *va_to_pte(unsigned long address);
 extern unsigned long ioremap_bot;
+
+#ifdef CONFIG_HIGHMEM
 extern void kunmap_virt(void *ptr);
+#else
+static inline void kunmap_virt(void *ptr)
+{
+}
+#endif
 
 #ifdef CONFIG_44x
 extern int icache_44x_need_flush;
-- 
1.7.0.4

